### PR TITLE
ci: Migrate Litmus and CalDAVTester from drone to GitHub

### LINKED ISSUE #41187: Move from Drone to GitHub Actions
* We then can reuse Drone workers more flexible as GitHub runners for other tasks
* No need to maintain two different CI systems
* Ability to restart failed CI processes without Drone admin rights

### Progress
* [x] Move all unit tests
  * https://github.com/nextcloud/server/pull/41003
* [x] Move all integration tests
  * [x] https://github.com/nextcloud/server/pull/44191
  * [x] https://github.com/nextcloud/server/pull/41139
  * [x] WebDav https://github.com/nextcloud/server/pull/43220
  * [x] Samba https://github.com/nextcloud/server/pull/43516
* [x] Move checkers https://github.com/nextcloud/server/pull/43514
* [ ] Move Litmus / CalDAV testsuite
* [x] Move acceptance tests (best case: use Cypress)
  * [x] Login tests: https://github.com/nextcloud/server/pull/41033
  * [x] Access levels: https://github.com/nextcloud/server/pull/41137
  * [x] Contacts menu: https://github.com/nextcloud/server/pull/41190
  * [x] Apps: https://github.com/nextcloud/server/pull/41303

We should prepare all of them and move them together so we can reuse the Drone workers quickly and reduce Drone to 3-5 workers for stable branches and Talk.

cc @skjnldsv @nickvergessen

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #41187 

## Summary
Migrate the last two remaining CI tests from drone to GitHub CI. Allowing us to shut down drone workers.

## TODO

- [x] Remove drone from the required checks

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by st3iny — COMMENTED** [review]
Awesome, looks good to me!

---

```
ERROR: There were 1 missing files:
.drone.yml
ERROR: Please remove or add those files again or inform the release team about those now files to be included or excluded from the release tar ball.
```

Perhaps removing the file from https://github.com/nextcloud/server/blob/master/build/files-checker.php is sufficient?

**Review by kesselb — APPROVED** [review]
Thank you :clap:

**Review by st3iny — APPROVED** [review]

File: .github/workflows/integration-caldav.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    if: needs.changes.outputs.src != 'false' && github.repository_owner != 'nextcloud-gmbh'
+
+    strategy:
+      # do not stop on another job's failure
+      fail-fast: false
+      matrix:
+        php-versions: ['8.3']
+        endpoint: ['old', 'new']
+        service: ['CalDAV', 'CardDAV']
```
Comment (nickvergessen) [inline .github/workflows/integration-caldav.yml:48]: so caldav integration test is also testing carddav? should we rename to dav?

File: .github/workflows/integration-litmus.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    permissions:
+      contents: none
+    runs-on: ubuntu-latest-low
+    needs: [changes, integration-litmus]
+
+    if: always()
+
+    steps:
+      - name: Summary status
+        run: if ${{ needs.changes.outputs.src != 'false' && needs.integration-litmus.result != 'success' }}; then exit 1; fi
```
Comment (nickvergessen) [inline .github/workflows/integration-litmus.yml:112]: missing new line

File: build/files-checker.php [hunk 3]
```diff
@@ -8,7 +8,6 @@
 	'.',
 	'..',
 	'.devcontainer',
-	'.drone.yml',
```
Comment (nickvergessen) [inline build/files-checker.php:11]: :tada:

**Review by nickvergessen — APPROVED** [review]

**Comment by susnux** [comment]
@nickvergessen I resolved you comments, when merging you need to remove *drone* from the required checks.

**Pull request merged**